android/src/main/java/com/rudderstack/android/RudderAnalytics.kt

```diff
@@ -70,8 +70,9 @@ private fun RudderAnalytics(
     }
 }
 
+private const val DEFAULT_INSTANCE = "default_instance"
```

Comment:
Why do we need this? We have the writekey, we can use it.

Reply:
If we use `writeKey` to map the instance (as we are doing in multiple instances) and the user makes another call with a different `writeKey` then, in that case, it'll try to create a new instance and break the singleton pattern, right?
In order to make sure `getInstance` always returns the same instance I've used the `DEFAULT_INSTANCE` logic.

Please let me know if I'm missing something.

Reply:
I agree with @itsdebs comment. Please check this out, i think it will help you create a proper singleton for Analytics class. 
[feat__creates_get_instance_method_to_preserve_singleton.patch](https://github.com/rudderlabs/rudder-sdk-android/files/15354668/feat__creates_get_instance_method_to_preserve_singleton.patch)

Reply:
`AnalyticsRegistry` can be utilised again, when multiple instances are going to be available.

Reply:
I have this concern that we might need to throw exception or log error if write key is different, but map is not empty. Let's discuss this once, then I can approve.

Reply:
@itsdebs I'm going to proceed with the patch approach for now. If you think we can improve it further then we can discuss it.